tests/build/test_flash.py

```diff
@@ -0,0 +1,139 @@
+from mbed_tools.build.exceptions import BinaryFileNotFoundError, BinaryFlashingFailedError, DeviceNotFoundError
+
+
+@mock.patch("mbed_tools.build.flash.get_connected_devices")
+@mock.patch("mbed_tools.build.flash.flash_dev")
+class TestFlashBinary(TestCase):
+    def test_check_binary_flashing(self, flash_dev, get_connected_devices):
+        mbed_board = Board(
+            board_type="TEST",
+            board_name="",
+            product_code="",
+            target_type="",
+            slug="",
+            build_variant=[],
+            mbed_os_support=[],
+            mbed_enabled=[],
+        )
+        test_device = Device(serial_port="", serial_number="", mount_points=[pathlib.Path(".")], mbed_board=mbed_board)
+
+        test_connected_devices = ConnectedDevices()
+        test_connected_devices.identified_devices = [test_device]
+        get_connected_devices.return_value = test_connected_devices
+
+        flash_dev.return_value = True
+
+        with tempfile.TemporaryDirectory() as tmpDir:
+            base_dir = pathlib.Path(tmpDir)
+            build_dir = base_dir / "cmake_build"
+            os.mkdir(build_dir)
+            bin_file = os.path.basename(base_dir) + ".bin"
+            bin_file = build_dir / bin_file
+            pathlib.Path(bin_file).touch()
+            flash_binary(base_dir, build_dir, "TEST", False)
+
+    def test_check_hex_flashing(self, flash_dev, get_connected_devices):
+        mbed_board = Board(
+            board_type="TEST",
+            board_name="",
+            product_code="",
+            target_type="",
+            slug="",
+            build_variant=[],
+            mbed_os_support=[],
+            mbed_enabled=[],
+        )
+        test_device = Device(serial_port="", serial_number="", mount_points=[pathlib.Path(".")], mbed_board=mbed_board)
+
+        test_connected_devices = ConnectedDevices()
+        test_connected_devices.identified_devices = [test_device]
+        get_connected_devices.return_value = test_connected_devices
+
+        flash_dev.return_value = True
+
+        with tempfile.TemporaryDirectory() as tmpDir:
+            base_dir = pathlib.Path(tmpDir)
+            build_dir = base_dir / "cmake_build"
+            os.mkdir(build_dir)
+            bin_file = os.path.basename(base_dir) + ".hex"
+            bin_file = build_dir / bin_file
+            pathlib.Path(bin_file).touch()
+            flash_binary(base_dir, build_dir, "TEST", True)
+
+    def test_check_flashing_failure(self, flash_dev, get_connected_devices):
+        mbed_board = Board(
+            board_type="TEST",
+            board_name="",
+            product_code="",
+            target_type="",
+            slug="",
+            build_variant=[],
+            mbed_os_support=[],
+            mbed_enabled=[],
+        )
+        test_device = Device(serial_port="", serial_number="", mount_points=[pathlib.Path(".")], mbed_board=mbed_board)
+
+        test_connected_devices = ConnectedDevices()
```

Comment:
The set up steps for building the `ConnectedDevices` object seem to be duplicated in multiple tests. We should factor this setup out into a factory function to reduce duplication.

Reply:
Implemented factory function as suggested.